Core/include/Acts/TrackFitting/GainMatrixUpdater.hpp

```diff
@@ -11,44 +11,32 @@
 #include "Acts/EventData/Measurement.hpp"
 #include "Acts/EventData/MeasurementHelpers.hpp"
 #include "Acts/EventData/MultiTrajectory.hpp"
-#include "Acts/EventData/TrackParameters.hpp"
+#include "Acts/Geometry/GeometryContext.hpp"
 #include "Acts/TrackFitting/KalmanFitterError.hpp"
-#include "Acts/TrackFitting/detail/VoidKalmanComponents.hpp"
-#include "Acts/Utilities/Definitions.hpp"
 #include "Acts/Utilities/Logger.hpp"
 #include "Acts/Utilities/Result.hpp"
 
-#include <memory>
-#include <variant>
-
 namespace Acts {
 
-/// @brief Update step of Kalman Filter using gain matrix formalism
+/// Kalman update step using the gain matrix formalism.
 class GainMatrixUpdater {
  public:
-  /// @brief Public call operator for the boost visitor pattern
-  ///
-  /// @tparam track_state_t Type of the track state for the update
-  ///
-  /// @param gctx The current geometry context object, e.g. alignment
-  /// @param trackState the measured track state
-  /// @param direction the navigation direction
+  /// Run the Kalman update step for a single trajectory state.
   ///
-  /// @return Bool indicating whether this update was 'successful'
-  /// @note Non-'successful' updates could be holes or outliers,
-  ///       which need to be treated differently in calling code.
-  template <typename track_state_t>
-  Result<void> operator()(const GeometryContext& /*gctx*/,
-                          track_state_t trackState,
-                          const NavigationDirection& direction = forward,
-                          LoggerWrapper logger = getDummyLogger()) const {
+  /// @tparam source_link_t The type of source link
+  /// @tparam kMeasurementSizeMax
+  /// @param[in] gctx The current geometry context object, e.g. alignment
+  /// @param[in,out] trackState The track state
+  /// @param[in] direction The navigation direction
+  /// @param[in] logger Where to write logging information to
+  template <typename source_link_t, size_t kMeasurementSizeMax>
```

Comment:
This has to be given in the client code then, right?

Reply:
No, this is automatically deduced from the arguments.

Reply:
ok